scripts/export_new_config_to_default_config.py

```diff
@@ -0,0 +1,247 @@
+
+    with DidFileEdit(toml_file, yaml_file) as check_file:
+        with open(toml_file, "w") as f:
+            f.write(MESSAGE.format(file_type="TOML"))
+            f.write("\n")
+            atoml.dump(doc, f)
+
+        with open(yaml_file, "w") as f:
+            f.write(MESSAGE.format(file_type="YAML"))
+            yaml.dump(dump, f, indent=4, Dumper=yaml.SafeDumper)
+
+    for file, diff in check_file.edited_files.items():
+        print(
+            f"Exported new configuration to {pathlib.Path(file).relative_to(MODMAIL_DIR.parent)}.",
+            file=sys.stderr,
+        )
+        try:
+            print("".join(diff))
+        except TypeError:
+            print("No diff to show.")
+        print()
+    return bool(len(check_file.edited_files))
+
+
+def export_env_and_app_json_conf() -> int:
+    """
+    Exports required configuration variables to .env.template.
+
+    Does NOT export *all* settable variables!
+
+    Export the *required* environment variables to `.env.template`.
+    Required environment variables are any Config.default variables that default to marshmallow.missing
+    These can also be configured by using the ConfigMetadata options.
+    """
+    default = modmail.config.get_default_config()
+
+    # find all environment variables to report
+    def get_env_vars(klass: type, env_prefix: str = None) -> typing.Dict[str, MetadataDict]:
+
+        if env_prefix is None:
+            env_prefix = modmail.config.ENV_PREFIX
+
+        # exact name, default value
+        export: typing.Dict[str, MetadataDict] = dict()  # any missing required vars provide as missing
+
+        for var in attr.fields(klass):
+            if attr.has(var.type):
+                # var is an attrs class too, run this on it.
+                export.update(
+                    get_env_vars(
+                        var.type,
+                        env_prefix=env_prefix + var.name.upper() + "_",
+                    )
+                )
+            else:
+                meta: MetadataDict = var.metadata
+                # put all values in the dict, we'll iterate through them later.
+                export[env_prefix + var.name.upper()] = meta
+
+        return export
+
+    with DidFileEdit(ENV_EXPORT_FILE, APP_JSON_FILE) as check_file:
+        # dotenv modifies currently existing files, but we want to erase the current file
+        ENV_EXPORT_FILE.unlink(missing_ok=True)
+        ENV_EXPORT_FILE.touch()
+
+        exported = get_env_vars(default.__class__)
+
+        with open(APP_JSON_FILE) as f:
+            try:
+                app_json: typing.Dict = json.load(f)
+            except Exception as e:
+                print(
+                    "Oops! Please ensure the app.json file is valid json! "
+                    "If you've made manual edits, you may want to revert them.",
+                    file=sys.stderr,
+                )
+                raise e
```

Comment:
We should open app_json before erasing the env export file, to avoid leaving files empty in case of errors.